src/core/multimedia/qgsvideoexporter.h

```diff
@@ -0,0 +1,147 @@
+/***************************************************************************
+                             qgsvideoexporter.h
+                             --------------------------
+    begin                : November 2025
+    copyright            : (C) 2025 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSVIDEOEXPORTER_H
+#define QGSVIDEOEXPORTER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+
+#include <QSize>
+#include <QObject>
+#include <QtMultimedia/QMediaFormat>
+#include <QtMultimedia/QMediaRecorder>
+
+class QMediaCaptureSession;
+class QVideoFrameInput;
+
+/**
+ * \ingroup core
+ * \brief Handles exports of sequential image files to video formats.
+ *
+ * Video export functionality is not available on all systems. The
+ * isAvailable() function can be used to test whether video
+ * export is available on the current system.
+ *
+ * \since QGIS 4.0
+ */
+class CORE_EXPORT QgsVideoExporter : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Returns TRUE if the video export functionality is available on the current system.
+     */
+    static bool isAvailable();
+
+    /**
+     * Constructor for QgsVideoExporter.
+     *
+     * \param filename destination video file name
+     * \param size output video frame size
+     * \param framesPerSecond output video frames per second
+     */
+    QgsVideoExporter( const QString &filename, QSize size, double framesPerSecond );
+    ~QgsVideoExporter() override;
+
+    /**
+     * Sets the list of input image \a files.
+     *
+     * The list must be an ordered list of existing image file paths, which will form
+     * the output video frames.
+     *
+     * \see inputFiles()
+     */
+    void setInputFiles( const QStringList &files );
```

Comment:
It could be useful to have a setInputFilesFromTemplate( QString directory, QString template )  too? Would fit in nicely with 

<img width="630" height="381" alt="image" src="https://github.com/user-attachments/assets/99d551fa-a6d3-4ec5-b2c6-242f5c68309d" />

and

<img width="630" height="381" alt="image" src="https://github.com/user-attachments/assets/c4853df7-abd3-4236-8c19-8338548f7192" />

Reply:
Just to be clear, we'd obviously update these dialogs to allow for a direct export to video, I'm just thinking about what would be useful to expose through the API itself :)

Reply:
I've added setInputFilesByPattern, which does similar